src/runtime/utils.ts

```diff
@@ -11,3 +13,45 @@ export const IS_BROWSER = typeof document !== "undefined";
 export function asset(path: string) {
   return `${INTERNAL_PREFIX}${STATIC_PREFIX}/${__FRSH_BUILD_ID}${path}`;
 }
+
+// apply the asset function a srcset
+export function assetSrcSet(srcset: string) {
+  const srcsetParts = srcset.split(",");
+  return srcsetParts.map((part) => {
+    const sections = part.trim().split(" ");
+    sections[0] = asset(sections[0]);
+    return sections.join(" ");
+  }).join(", ");
+}
+
+export function assetHashingHook(vnode: VNode) {
+  if (vnode.type === "img" || vnode.type === "source") {
+    const props = (vnode.props as HTMLImageElement);
+    // deno-lint-ignore no-explicit-any
+    if ((props as any)["data-no-auto-hashing"]) {
+      // skip
```

Comment:
This should just be an early return, you can skip the next else and reduce the nesting